### PR TITLE
Oauth/manager/introduce ticks

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: security/oauth/manager/manager.go [hunk 1]
```diff
@@ -63,48 +77,50 @@ func (a *Manager) Close() {
...
-	t.TLSClientConfig = tlsCfg
-	httpClient := &http.Client{Transport: t}
 	ctx = context.WithValue(ctx, oauth2.HTTPClient, httpClient)
 
-	return cfg.Token(ctx)
+	token, err := cfg.Token(ctx)
+	var startRefreshToken time.Time
+	if err == nil {
+		now := time.Now()
+		startRefreshToken = now.Add(token.Expiry.Sub(now) * 2 / 3)
```
[the remark this answers is not in the document]
Comment (jkralik) [reply security/oauth/manager/manager.go:94]: Yes

File: security/oauth/manager/config.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ type Config struct {
 	Endpoint       Endpoint      `envconfig:"ENDPOINT" env:"ENDPOINT"`
 	Audience       string        `envconfig:"AUDIENCE" env:"AUDIENCE"`
 	RequestTimeout time.Duration `envconfig:"REQUEST_TIMEOUT" env:"REQUEST_TIMEOUT" default:"10s"`
+	TickFrequency  time.Duration `envconfig:"TICK_FREQUENCY" env:"TICK_FREQUENCY" long:"tick-frequency" description:"how frequently we should check whether our token needs renewal" default:"15s"`
```
[the remark this answers is not in the document]
Comment (jkralik) [reply security/oauth/manager/config.go:21]: Richt. The TickFrequency  I just used it in another configuration. Do you have better name ?

[the remark this answers is not in the document]
Comment (der-dano) [reply security/oauth/manager/config.go:21]: you included the description, so its good :)

File: security/oauth/manager/manager.go [hunk 3]
```diff
@@ -63,48 +77,50 @@ func (a *Manager) Close() {
...
 
-func (a *Manager) nextRenewal() time.Duration {
-	t, _ := a.GetToken(context.Background())
-	now := time.Now()
-	lifetime := t.Expiry.Sub(now) * 2 / 3
-	if lifetime < a.requestTimeout {
-		lifetime = a.requestTimeout
-	}
-	return lifetime
+func (a *Manager) wantToRefresh() bool {
```
[the remark this answers is not in the document]
Comment (jkralik) [reply security/oauth/manager/manager.go:80]: Renamed

**Review by der-dano — APPROVED** [review]

**Pull request merged**